src/lib/GifRequestDispatcher.ts

```diff
@@ -0,0 +1,177 @@
+                    setting.message
+                );
+                return false;
+            }
+
+            if (
+                setting.key === Preferences.WEBHOOK_URL ||
+                setting.key === Preferences.API_URL
+            ) {
+                try {
+                    new URL(setting.value);
+                } catch (e) {
+                    const errorMessage = `Invalid URL assigned to ${setting.key}: ${setting.value}`;
+                    this.app.getLogger().log(errorMessage);
+                    sendMessageToSelf(
+                        this.modify,
+                        this.room,
+                        this.sender,
+                        this.threadId,
+                        errorMessage
+                    );
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    async generateGif(prompt: string): Promise<IGifResponseData | Error> {
+        const apiUrl = await getSettingFromId(this.read, Preferences.API_URL);
+        const apiKey = await getSettingFromId(this.read, Preferences.API_KEY);
+        const modelId = await getSettingFromId(this.read, Preferences.MODEL_ID);
+        const webhookUrl = await getSettingFromId(
+            this.read,
+            Preferences.WEBHOOK_URL
+        );
+
+        if (!apiKey || !apiUrl || !modelId || !webhookUrl) {
+            const errorMessage =
+                "ValidationError: One or more preferences are not set";
+            this.app.getLogger().log(errorMessage);
+            throw new Error(errorMessage);
+        }
+
+        const requestBody: IGifRequestBody = {
+            version: modelId,
+            webhook: webhookUrl,
+            input: {
+                mp4: false,
+                steps: 30,
+                width: 672,
+                height: 384,
+                prompt,
+                negative_prompt: "blurry",
+            },
+        };
+
+        const res = await this.http.post(apiUrl, {
+            headers: {
+                Authorization: `Bearer ${apiKey}`,
+                "Content-Type": "application/json",
+            },
+            data: requestBody,
+        });
+
+        if (!res || !res.data || res.statusCode !== 201) {
+            const responseData = res.data as IGifResponseData;
+
+            if (
+                responseData.status === PredictionStatus.FAILED ||
+                responseData.status === PredictionStatus.CANCELLED
+            ) {
+                this.app.getLogger().log(responseData.error);
+                throw new Error(responseData.error);
+            }
+        }
+
+        return res.data as IGifResponseData;
+    }
+
+    async mockGenerateGif(
+        prompt: string,
+        id: string
+    ): Promise<IGifResponseData> {
+        const webhookUrl = await getSettingFromId(
+            this.read,
+            Preferences.WEBHOOK_URL
+        );
+
+        console.log("Mocking generation of gif", webhookUrl);
```

Comment:
Can you remove this console.log, never push console.log

Reply:
Sorry missed this out! fixing this real quick